### PR TITLE
Implement extensions to allow for MOAT

### PR DESCRIPTION
**Pull request by darynl**
# Context

This PR adds the extension for MOAT.

# Acceptance

This does not have negative impacts on production.

# Changes

* Added the extension AdVerifications>Verifications> ViewableImpression which MOAT uses for placing trackers.

### CODE REVIEW

File: vast/extension.go [hunk 1]
```diff
@@ -3,5 +3,16 @@ package vast
...
 // platform to extend the VAST response.
 type Extension struct {
-	Data []byte `xml:",innerxml"`
+	Data          []byte          `xml:",innerxml"`
+	Verifications []*Verification `xml:"AdVerifications>Verification,omitempty"`
+}
+
+type Verification struct {
+	Vendor             string              `xml:"vendor,attr"`
+	ViewableImpression *ViewableImpression `xml:"ViewableImpression"`
```
Comment (gootik) [inline vast/extension.go]: I think this needs to be an array of impressions...  there can be multiple `ViewableImpression` tags in the VAST

Comment (gootik) [reply vast/extension.go]: oh my bad, I read it incorrectly. this is correct

File: vast/extension.go [hunk 2]
```diff
@@ -3,5 +3,16 @@ package vast
 // Extension type represents an <Extension> element that contains arbitrary XML data provided by the
 // platform to extend the VAST response.
 type Extension struct {
-	Data []byte `xml:",innerxml"`
+	Data          []byte          `xml:",innerxml"`
```
Comment (darynl) [inline vast/extension.go]: Do we need this field? It's causing test issues with marshalling then unmarshalling.

Comment (hwong) [reply vast/extension.go]: Doesn't look like it's used anywhere.

**Comment by darynl** [comment]
Closed due to constrained functionality of implementation. In order to serve the needs of all users of this library, we decided to keep the Extension field open to individual needs.

**Pull request closed without merging**